src/pages/rest/authentication/user.md

```diff
@@ -0,0 +1,172 @@
+  1. Add the **Adobe Commerce with Adobe ID** API to your project to enable Commerce-specific authentication.
+
+- Creating an OAuth User authentication project:
+
+  1. Select the preferred OAuth 2 authentication type: **Web App/Single-Page App/Native App**.
+  1. Configure the allowed redirect URIs.
+  1. Copy the Client ID and Client Secret.
+  1. Make a note of the authorized redirect URIs.
+  1. Securely save your credentials.
+
+### 2. Authorization flow
+
+**Building authorization URL**
+
+The authorization URL is used to initiate the authentication process. It includes the client ID, redirect URI, scopes, and a state parameter for security. Here is the example for a web app:
+
+```html
+https://ims-na1.adobelogin.com/ims/authorize/v2?client_id={{client_id}}&redirect_uri={{redirect_uri}}&scope={{scopes}}&state=something&response_type=code
+```
+
+Replace the following placeholders with your values:
+
+- `{{client_id}}`: IMS client ID
+- `{{redirect_uri}}`: Configured redirect URI
+- `{{scopes}}`: Coma-separated list of required scopes:
+
+  ```bash
+  AdobeID,openid,email,profile,additional_info.roles,additional_info.projectedProductContext`
+  ```
+
+**Handling authorization response**
+
+- Redirect handling:
+  1. User completes authentication
+  1. Browser redirects to your `redirect_uri`
+  1. Authorization code is included in URL parameters
+- Authorization code extraction:
+  1. Parse code from URL: `?code={{auth_code}}&state=something`
+  1. Verify state parameter matches original request
+- Error handling:
+  1. Check for error parameters in redirect
+  1. Implement appropriate error messaging
+  1. Provide retry mechanisms
+
+### 3. Token exchange
+
+- Authorization code to access token:
+  1. Make a `POST` request to the token endpoint.
+
+<CodeBlock slots="heading, code" repeat="2" languages="HTTP, JSON" />
+
+#### Request
+
+```http
+POST https://ims-na1.adobelogin.com/ims/token/v3
+Authorization: Basic {{base64(client_id:client_secret)}}
+Content-Type: application/x-www-form-urlencoded
+
+code={{auth_code}}&grant_type=authorization_code
+```
+
+#### Response:
+
+```json
+{
+    "access_token": "{ACCESS_TOKEN}",
+    "refresh_token": "{REFRESH_TOKEN}",
+    "sub": "A0BC123D4CD449CA0A494133@a12b34cd5b5b7e0e0a494004",
+    "id_token": "{ID_TOKEN}",
+    "token_type": "bearer",
+    "expires_in": 86399
+}
+```
+
+- Token refresh process:
+  1. Monitor token expiration.
+  1. Use refresh token to obtain new access token.
+
+<CodeBlock slots="heading, code" repeat="2" languages="HTTP, JSON" />
+
+#### Request
+
+```http
+POST https://ims-na1.adobelogin.com/ims/token/v3
+Authorization: Basic {{base64(client_id:client_secret)}}
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=refresh_token&refresh_token={{refresh_token}}
+```
+
+#### Response
+
+```json
+{
+"access_token": "{ACCESS_TOKEN}",
+"refresh_token": "{REFRESH_TOKEN}",
+"expires_in": 86399,
+"token_type": "bearer"
+}
+```
+
+  1. Update the stored tokens.
```

Comment:
This ordered list item will not continue numbering because the example above breaks the order. You could switch these `request`/`response` examples to `_includes` but for speed, maybe just move the example to after the list.